code/modules/mob/living/carbon/human/species/station/machine_subspecies.dm

```diff
@@ -0,0 +1,228 @@
+/datum/species/machine/shell
+	name = "Shell Frame"
+	short_name = "jak"
+	name_plural = "Shells"
+
+	burn_mod = 1.2
+
+	blurb = "IPC's with humanlike properties. Their focus is on service, civilian, and medical, but there are no \
+	job restrictions. Created in the late days of 2457, the Shell is a controversial IPC model equipped with a synthskin weave applied over its metal chassis \
+	to create an uncannily close approximation of the organic form. Early models of Shell had the advantage of being able to compose themselves of a wide \
+	 variety of organic parts, but contemporary models have been restricted to a single species for the sake of prosthetic integrity. The additional weight of \
+	 the synthskin on the original Hephaestus frame reduces the efficacy of the IPC's already strained coolant systems, and increase charge consumption."
+
+	num_alternate_languages = 3
+
+	icobase = 'icons/mob/human_races/robotic.dmi'
+	deform = 'icons/mob/human_races/robotic.dmi'
+
+	eyes = "eyes_s"
+	show_ssd = "completely quiescent"
+
+	max_nutrition_factor = 0.8
+
+	heat_level_1 = 400
+	heat_level_2 = 800
+	heat_level_3 = 1600
+
+	heat_discomfort_level = 400
+	heat_discomfort_strings = list(
+		"Your CPU temperature probes warn you that you are approaching critical heat levels!",
+		"Your synthetic flesh crawls in the heat, swelling into a disgusting morass of plastic."
+		)
+
+	appearance_flags = HAS_HAIR_COLOR | HAS_SKIN_TONE | HAS_EYE_COLOR |  HAS_SKIN_COLOR | HAS_FBP
+
+/datum/species/machine/shell/handle_post_spawn(var/mob/living/carbon/human/H)
+	return ..()
+
+/datum/species/machine/shell/handle_sprint_cost(var/mob/living/carbon/human/H, var/cost)
+	if (H.stat == CONSCIOUS)
+		H.bodytemperature += cost*1.3
+		H.nutrition -= cost*0.85
+		if (H.nutrition > 0)
+			return 1
+		else
+			H.Weaken(30)
+			H.m_intent = "walk"
+			H.hud_used.move_intent.update_move_icon(H)
+			H << span("danger", "ERROR: Power reserves depleted, emergency shutdown engaged. Backup power will come online in 60 seconds, initiate charging as primary directive.")
+			playsound(H.loc, 'sound/machines/buzz-two.ogg', 100, 0)
+	return 0
+
+/datum/species/machine/shell/handle_death(var/mob/living/carbon/human/H)
```

Comment:
If a proc simply calls `..()` without additional actions, you can delete its definition.